File: src/main/java/leetcode/lists/offerII/Offer54.java
package leetcode.lists.offerII;

import common.*;

/**
 * @author wujingxinit@outlook.com
 * @date 2023/7/25 17:10
 */
public class Offer54 {

    int sum = 0;

    public TreeNode convertBST(TreeNode root) {
        if (root != null) {
            convertBST(root.right);
            sum += root.val;
            root.val = sum;
            convertBST(root.left);
        }
        return root;
    }

}

File: go/src/leetcode/offerII/Offer54.go
package offerII

/**
 * @author wujingxinit@outlook.com
 * @date 2023/1/17 18:30
 */

func convertBST(root *TreeNode) *TreeNode {

	if root == nil {
		return nil
	}
	sum := 0
	var dfs func(root *TreeNode)
	dfs = func(root *TreeNode) {
		if root == nil {
			return
		}
		dfs(root.Right)
		//op
		root.Val += sum
		sum = root.Val
		dfs(root.Left)
	}
	dfs(root)
	return root
}
